imapclient/imapclient.py

```diff
@@ -162,9 +169,22 @@ def __init__(self, host, port=None, use_uid=True, ssl=True, stream=False,
         self._imap.debug = 5
         self._imap._mesg = imaplib_logger.debug
 
-        self._idle_tag = None
+    def __enter__(self):
+        return self
 
-        self._set_timeout()
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        """Logout and closes the connection when exiting the context manager.
+
+        All exceptions are caught because an error in `logout` or `shutdown`
```

Comment:
"All exceptions during logout and connection shutdown are caught because an error here usually means the connection was already closed."